frontend-react/src/utils/OrganizationUtils.ts

```diff
@@ -96,9 +56,55 @@ export const groupToOrg = (group: string | undefined): string => {
     }
 };
 
-export function useOrgName(): string {
-    const org = useResource(OrganizationResource.detail(), {
-        name: groupToOrg(getStoredOrg()),
+/* Delivers the array of organization names, or the first, parsed from Okta
+ * groups. */
+const getRSOrgs = (
+    token: AccessToken | undefined,
+    onlyType?: RSOrgType
+): string[] | string => {
+    if (!token || !token.claims) return [];
+    /* Have to convert to RSUserClaims type to get Organization claim */
+    const oktaGroups = getOktaGroups(token);
+
+    /* Consolidate all the filtering logic */
+    const filterGroups = (): string[] => {
+        switch (onlyType) {
+            /* Return those WITH sender */
+            case RSOrgType.SENDER:
+                return oktaGroups.filter((group) => isSender(group));
+            /* Return those WITHOUT sender */
+            case RSOrgType.RECEIVER:
+                return oktaGroups.filter((group) => isReceiver(group));
+            /* Return ONLY admin groups */
+            case RSOrgType.ADMIN:
+                return oktaGroups.filter((group) => isAdmin(group));
+            /* Return all groups */
+            default:
+                return oktaGroups;
+        }
+    };
+    return filterGroups().map((group) => groupToOrg(group));
+};
+
+function parseOrgs(orgs: Array<string>): Array<Partial<SessionStore>> {
+    return orgs.map((org) => {
+        // Org names are case sensitive. This condition will fail if the okta
+        // group name is not cased properly: DHSender_xyz, DHxy_phd, DHPrimeAdmin
+        if (org.includes(PERMISSIONS.SENDER)) {
+            const sender = org.split(".");
+            return {
+                org: groupToOrg(sender[0]),
+                senderName: sender[1] || "default",
+            };
+        } else {
+            return {
+                org: groupToOrg(org),
+                senderName: undefined,
+            };
+        }
     });
-    return org?.description || "";
 }
+
+export { RSOrgType, getOktaGroups, groupToOrg, getRSOrgs, parseOrgs };
```

Comment:
Why export this way instead of just putting `export` in front of each declaration?

Reply:
I found this was a good exercise for me to determine what _really_ needed to be exported and why. It also is a nice way to take census of what a module is exporting vs what's internal when browsing at a glance!